File: sponge-examples/examples/script/js/knowledge_base_load2.js
/**
 * Sponge Knowledge base
 * Loading knowledge bases
 */

var Trigger1 = Java.extend(Trigger, {
    configure: function(self) {
        self.displayName = "Trigger1, file2";
        self.eventName = "e1";
    },
    run: function(self, event) {
        self.logger.debug("file2: Received event {}", event);
        EPS.getVariable("eventCounter").get(self.displayName).incrementAndGet();
    }
});

var Trigger2 = Java.extend(Trigger, {
    configure: function(self) {
        self.displayName = "Trigger2, file2";
        self.eventName = "e2";
    },
    run: function(self, event) {
        self.logger.debug("file2: Received event {}", event);
        EPS.getVariable("eventCounter").get(self.displayName).incrementAndGet();
    }
});

// Execute immediately while loading
EPS.enableAll(Trigger1, Trigger2);

function onShutdown() {
    EPS.logger.debug("onShutdown, file2");
}

File: sponge-jython/examples/script/py/knowledge_base_load2.py
"""
Sponge Knowledge base
Loading knowledge bases
"""

class Trigger1(Trigger):
    def onConfigure(self):
        self.withLabel("Trigger1, file2").withEvent("e1")
    def onRun(self, event):
        #self.logger.debug("file2: Received event {}", event)
        global eventCounter
        eventCounter.get(self.meta.label).incrementAndGet()


class Trigger2(Trigger):
    def onConfigure(self):
        self.withLabel("Trigger2, file2").withEvent("e2")
    def onRun(self, event):
        #self.logger.debug("file2: Received event {}", event)
        global eventCounter
        eventCounter.get(self.meta.label).incrementAndGet()

# Execute immediately while loading
sponge.enableAll(Trigger1, Trigger2)

def onShutdown():
    sponge.logger.debug("onShutdown, file2")







